packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -0,0 +1,891 @@
+
+        if (freeSlot) {
+          this.slots.video.activeSpeaker.push(freeSlot);
+          this.receiveSlotAllocations.activeSpeaker[group.id].slots.push(freeSlot);
+        }
+      }
+    });
+
+    this.currentLayout?.memberVideoPanes?.forEach((memberPane) => {
+      // check if there is existing slot for this csi
+      const existingSlot = this.slots.video.receiverSelected.find(
+        (slot) => slot.csi === memberPane.csi
+      );
+
+      if (memberPane.csi !== undefined && existingSlot) {
+        // found it, so use it
+        this.receiveSlotAllocations.receiverSelected[memberPane.id] = existingSlot;
+      } else {
+        // allocate a slot from the "unused" list
+        const freeSlot = this.slots.video.unused.pop();
+
+        if (freeSlot) {
+          this.slots.video.receiverSelected.push(freeSlot);
+          this.receiveSlotAllocations.receiverSelected[memberPane.id] = freeSlot;
+        }
+      }
+    });
+  }
+
+  /**
+   * Makes sure we have the right number of receive slots created for the current layout
+   * and allocates them to the right video pane groups
+   *
+   * @returns {Promise}
+   */
+  private async updateReceiveSlots() {
+    const requiredNumSlots = this.getRequiredNumVideoSlotsForLayout(this.currentLayout);
+    const totalNumSlots =
+      this.slots.video.unused.length +
+      this.slots.video.activeSpeaker.length +
+      this.slots.video.receiverSelected.length;
+
+    // ensure we have enough total slots for current layout
+    if (totalNumSlots < requiredNumSlots) {
+      let numSlotsToCreate = requiredNumSlots - totalNumSlots;
+
+      while (numSlotsToCreate > 0) {
+        // eslint-disable-next-line no-await-in-loop
+        this.slots.video.unused.push(
+          await this.receiveSlotManager.allocateSlot(MC.MediaType.VideoMain)
+        );
+        numSlotsToCreate -= 1;
+      }
+    }
+
+    // move all no longer needed receiver-selected slots to "unused"
+    this.trimReceiverSelectedSlots();
+
+    // move all active speaker slots to "unused"
+    this.slots.video.unused.push(...this.slots.video.activeSpeaker);
+    this.slots.video.activeSpeaker.length = 0;
+
+    // allocate the slots to the right pane groups
+    this.allocateSlotsToVideoPaneGroups();
+
+    LoggerProxy.logger.log(
+      `RemoteMediaManager#updateReceiveSlots --> receive slots updated: unused=${this.slots.video.unused.length}, activeSpeaker=${this.slots.video.activeSpeaker.length}, receiverSelected=${this.slots.video.receiverSelected.length}`
+    );
+
+    // If this is the initial layout, there may be some "unused" slots left because of the preallocation
+    // done in this.preallocateVideoReceiveSlots(), so release them now
+    this.releaseUnusedVideoSlots();
+  }
+
+  /**
+   * Creates new RemoteMedia and RemoteMediaGroup objects for the current layout
+   * and sends the media requests for all of them.
+   */
+  private updateRemoteMediaObjects() {
```

Comment:
Would be clearer to name it updateVideoRemoteMediaObjects

Reply:
done